libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/OAuth/CSFSalesforceOAuthRefresh.m

```diff
@@ -97,6 +97,12 @@ - (void)oauthCoordinatorDidAuthenticate:(SFOAuthCoordinator *)coordinator authIn
     self.network.account.credentials = coordinator.credentials;
     CSFOAuthTokenRefreshOutput *output = [[CSFOAuthTokenRefreshOutput alloc] initWithCoordinator:coordinator];
     [self finishWithOutput:output error:nil];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[NSNotificationCenter defaultCenter] postNotificationName:kSFAuthenticationManagerFinishedNotification
```

Comment:
+1.  I don't see an immediate reason why this notification needs to be delivered via the main thread.